lib/engine/train.rb

```diff
@@ -78,6 +79,18 @@ def buyable
       @buyable && !@obsolete
     end
 
+    def local?
+      return @local unless @local.nil?
+
+      @local = false
+      if @distance.is_a?(Numeric)
+        @local = @distance == 1
+      else
+        distance_city = @distance.find { |n| n['nodes'].include?('city') }
+        @local = distance_city['visit'] == 1 if distance_city
+      end
+    end
+
```

Comment:
Is there a reason that @local shouldn't be determined at initialization?

Reply:
No reason, i can do an init_local if that fits better.

Reply:
Yeah, that would be better. Then it can simply be an attr_reader var.